packages/plugin-support/src/support.js

```diff
@@ -0,0 +1,174 @@
+/**!
+ *
+ * Copyright (c) 2015-2016 Cisco Systems, Inc. See LICENSE file.
+ * @private
+ */
+
+import {SparkPlugin} from '@ciscospark/spark-core';
+import {defaults} from 'lodash';
+import uuid from 'uuid';
+
+
+const Support = SparkPlugin.extend({
+  namespace: `Support`,
+
+  _constructFileMetadata(metadata) {
+    let metadataArray = [];
+    let client = this.spark.client || {};
+    let trackingId = client.trackingIdBase;
+
+    if (this.spark.config.trackingIdPrefix) {
+      trackingId = `${this.spark.config.trackingIdPrefix}_${client.trackingIdBase}`;
+    }
+
+    [`locusId`, `callStart`, `feedbackId`].forEach(function addKeyToArray(key) {
+      if (metadata[key]) {
+        metadataArray.push({
+          key: key,
+          value: metadata[key]
+        });
+      }
+    });
+
+    metadataArray.push({
+      key: `trackingId`,
+      value: trackingId
+    });
+
+    return metadataArray;
+  },
+
+  getFeedbackUrl(options) {
+    options = options || {};
+    return this.request({
+      method: `POST`,
+      api: `conversation`,
+      resource: `users/deskFeedbackUrl`,
+      body: defaults(options, {
+        appVersion: this.config.appVersion,
+        appType: this.config.appType,
+        feedbackId: options.feedbackId || uuid.v4(),
+        languageCode: this.config.languageCode
+      })
+    })
+      .then((res) => res.body.url);
+  },
+
+  getSupportUrl() {
+    return this.spark.request({
+      method: `GET`,
+      api: `conversation`,
+      resource: `users/deskSupportUrl`,
+      qs: {
+        languageCode: this.config.languageCode
+      }
+    })
+      .then((res) => res.body.url);
+  },
+
+  submitCallLogs(metadata, logs) {
```

Comment:
Let's see if we can combine the two log methods into a single method called "submitLogs()" and have it decide to use client credentials if spark.isAuthorized (or isAuthenticated, or whatever that boolean is called) is false.